File: backend/src/ECommerce.Domain/Entities/Order.cs
public class Order
{

    public int Id { get; set; }
    public required string UserId { get; set; }
    public DateTime OrderDate { get; set; }
    public OrderStatus Status { get; set; }

    public decimal TotalAmount { get; set; }

    public string? StripeSessionId { get; set; }

    public required string ShippingFullName { get; set; }
    public required string ShippingPhoneNumber { get; set; }
    public required string ShippingStreet { get; set; }
    public required string ShippingCity { get; set; }
    public required string ShippingState { get; set; }
    public required string ShippingPostalCode { get; set; }
    public ICollection<OrderItem> OrderItems { get; set; } = new List<OrderItem>();
}

File: frontend/src/types/Order.ts
export interface OrderItem {
  id: number;
  productId: number;
  productTitle: string;
  unitPrice: number;
  quantity: number;
  lineTotal: number;
}

export interface Order {
  id: number;
  userId: string;
  orderDate: string;
  status: string;
  totalAmount: number;
  shippingFullName: string;
  shippingPhoneNumber: string;
  shippingStreet: string;
  shippingCity: string;
  shippingState: string;
  shippingPostalCode: string;
  items: OrderItem[];
}
